docs/examples/101/basic-batch-account/main.json

```diff
@@ -0,0 +1,37 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "batchAccountName": {
+      "type": "string"
+    },
+    "allocationMode": {
+      "type": "string",
+      "defaultValue": "BatchService",
+      "allowedValues": ["BatchService", "Usersubscription"]
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]"
+    }
+  },
+  "functions": [],
+  "variables": {},
```

Comment:
I had to rerun `bicep build` to get rid of this line which was the reason your test was failing. Ensure you are using bicep v0.2.212 as we have updated the ARM Template generation logic. A bicep file with no `vars` should result in an ARM Template with no `"variables"`